fixes.py

```diff
@@ -218,6 +224,10 @@ def metadata_fix(self):
             self.item_report['metadata_result'] = f'Metadata too long to upload from "{fc_path}" (>32,767 characters)'
             return
 
+        finally:
+            if arcpy.Exists(metadata_xml_path):
+                arcpy.management.Delete(metadata_xml_path)
```

Comment:
**nitpick**: this is just a path on disk and not spatial data in a gdb or anything, use pathlib to check for [existence](https://docs.python.org/3/library/pathlib.html#pathlib.Path.exists) and [removal](https://docs.python.org/3/library/pathlib.html#pathlib.Path.unlink).

Since you'll have the path object created above, it will be very easy to do so.

I think the take away is to stop using `os.path.join` and use arcpy are sparingly as possible.

Reply:
Ok, I've done as requested. 

Help me understand why it's worth the time to change (and test) an entire project based on a nitpick for a feature on a single part. Because we have to touch arcpy and arcpy doesn't do pathlib, it seems like it would make code more consistent to stick with string representations of paths everywhere rather than casting Path's to strings just for the arcpy tools. os.path still works, so I struggle to see the massive benefit to pathlib when we still have to cast to string.

Reply:
FWIW - I like pathlib better even with the pain of having to cast to strings for arcpy. I hope that arcpy will support path objects natively in the future. @steveoh may have some inside info on this. I think that it's just a direction that we'd like to go in the future. There are benefits to standardizing as much as possible as an office on these types of things. But yes, it can be a pain.